src/estimagic/benchmarking/process_benchmark_results.py

```diff
@@ -45,267 +42,146 @@ def create_convergence_histories(
-        target_values (pandas.Series): index are problems, values are target values.
-
-    Returns:
-        pandas.Series: index is the same as that of sr. The lower the value the closer
-            the current value is to the target value. 0 means the target value has been
-            reached. 1 means the current value is as far from the target value as the
-            start value.
-
-    """
-    # expand start and target values to the length of the full DataFrame
-    start_values = df["problem"].map(start_values)
-    target_values = df["problem"].map(target_values)
-
-    normalized = (df[col] - target_values) / (start_values - target_values)
-    return normalized
-
-
-def _clip_histories(df, stopping_criterion, x_precision, y_precision):
-    """Shorten the DataFrame to just the evaluations until each algorithm converged.
-
-    Args:
-        df (pandas.DataFrame): index levels are ['problem', 'algorithm', 'evaluation'].
-            Columns must include "monotone_criterion_normalized" if stopping_criterion
-            includes y and "monotone_parameter_distance_normalized" if x is in
-            the stopping_criterion.
-        stopping_criterion (str): one of "x_and_y", "x_or_y", "x", "y".
-        x_precision (float): when an algorithm's parameters are closer than this to the
-            true solution's parameters, the algorithm is counted as having converged.
-        y_precision (float): when an algorithm's criterion value is closer than this to
-            the solution value, the algorithm is counted as having converged.
-
-    Returns:
-        shortened (pandas.DataFrame): the entered DataFrame with all histories
-            shortened to stop once conversion according to the given criteria is
-            reached.
-        converged_info (pandas.DataFrame): columns are the algorithms, index are the
-            problems. The values are boolean and True when the algorithm arrived at
-            the solution with the desired precision.
-
-    """
-    # drop problems with no known solution
-    if "x" in stopping_criterion:
-        df = df[df["monotone_parameter_distance_normalized"].notnull()]
-    if "y" in stopping_criterion:
-        df = df[df["monotone_criterion_normalized"].notnull()]
-
-    # determine convergence in the known problems
-    if "x" in stopping_criterion:
-        x_converged = df["monotone_parameter_distance_normalized"] < x_precision
-    if "y" in stopping_criterion:
-        y_converged = df["monotone_criterion_normalized"] < y_precision
-
-    # determine converged function evaluations
-    if stopping_criterion == "y":
-        converged = y_converged
-    elif stopping_criterion == "x":
-        converged = x_converged
-    elif stopping_criterion == "x_and_y":
-        converged = y_converged & x_converged
-    elif stopping_criterion == "x_or_y":
-        converged = y_converged | x_converged
+def _aggregate_idxs_with_or(x, y):
+    if x is None or y is None:
+        out = None
```

Comment:
This should be `and` and then the else clause needs to be two elif clauses.

Reply:
Done.